core/includes/bootstrap.inc

```diff
@@ -1011,6 +1016,7 @@ function backdrop_get_filename($type, $name, $filename = NULL) {
  * @deprecated since 1.0
  */
 function variable_initialize($conf = array()) {
+  watchdog_deprecated_function('bootstrap', __FUNCTION__);
```

Comment:
Yeah I would prefer not to depend on a fluke of the module invoking system. Calling hooks early in bootstrap can lead to other issues, such as the module_implements() cache getting filled incorrectly.

> I have no objection to replacing that watchdog_deprecated_function() call with a comment.

We already mark deprecated functions with the `@deprecated` flag, which should suffice in this case.